### DISCUSSION TITLE
Welcome to Discussions!

### DISCUSSION THREAD

**Discussion by AnanyaJha** [forcedotcom/Einstein-GPT-for-Developers#4] [Announcements]
Welcome to Discussions! We're using Discussions as a place to chat about the Einstein GPT for Developers product and experience. We hope that you:

- [Ask questions you're curious about](https://github.com/forcedotcom/Einstein-GPT-for-Developers/discussions/categories/q-a)
- [Share new ideas](https://github.com/forcedotcom/Einstein-GPT-for-Developers/discussions/categories/ideas)
- [Discuss new features](https://github.com/forcedotcom/Einstein-GPT-for-Developers/discussions/categories/ideas)
- [Engage with Einstein GPT for Developers team members](https://github.com/forcedotcom/Einstein-GPT-for-Developers/discussions/categories/general)

You should still use issues when submitting bug reports and feature requests. Don't worry about posting in the wrong place - issues and discussions can be converted into one another and the Maintainers will do it whenever necessary. We'll slowly start migrating posts over from Issues as needed. Let's learn more about the Einstein GPT for Developers product and how y'all would like to use it, together! 💪

**Comment by vromanov-sumup** [comment]
Hey @AnanyaJha , I don't see the button to create a new discussion, could you check permissions on the repo?

**Comment by vromanov-sumup** [comment]
Posting Question here in the meantime: can we use Einstein GPT in Code Builder? 
when installing the extension in the Code Builder I see error "Unable to install extension 'salesforce.salesforcedx-einstein-gpt' as it is not compatible with VS Code '1.75.1'."
Is is possible to enable the extension for this version of Code Builder? When can we expect the version upgrade on the Code Builder itself?

**Reply by AnanyaJha** [reply]
Hi @vromanov-sumup thanks for the heads up about Discussions, working on enabling :) For Code Builder, we should be seeing that in the next few weeks - I'll get back to you with a concrete estimate asap

**Comment by samirbavkar** [comment]
Hello,   I have an installation related question? I do not see the "add" button to create a new discussion, so I am posting my question here.

Per the install instructions I installed the 3 vsix files:

- salesforcedx-einstein-gpt-0.14.0.vsix
- salesforcedx-vscode-apex-58.4.1.vsix
- salesforcedx-vscode-code-58.4.1.vsix

However, I do not see the  "SFDX: Generate Code with Einstein GPT"in the Command Palette. I do see the other 2 options "SFDX: Show GPT Prompt History" and "SFDX: Toggle Einstein GPT for Developers".

I have verified that the "Einstein GPT" extension is installed. Also, I have the latest Visual Studio Code ( Version:1.81.1). 

I am most likely missing a step.  Any advise on this will be very much appreciated. 

Thank you!

**Reply by abrantes23** [reply]
Hi @samirbavkar, I suspect the permission of Einstein GPT for Developers is not enabled on the org you are testing (you can test for example to switch to production on VS code). Apart from that, make sure your installed extensions are up-to-date.

**Reply by abrantes23** [reply]
You can check this, finding the einstein logo below on the right on your visual studio code.

Without permissions active:
![image](https://github.com/forcedotcom/Einstein-GPT-for-Developers/assets/135630762/0443f358-5f9f-47f3-83e8-71c6b6d6bff9)


With permissions active:
![image](https://github.com/forcedotcom/Einstein-GPT-for-Developers/assets/135630762/03aafa45-e722-41ad-a465-d18fa0750e43)

**Comment by abrantes23** [comment]
Code builder + Einstein steps:
- Code Builder installed on sandbox
- Created a new env from sandbox. Code Builder code-server: v4.16.0 Code: 1.80.1
- Make sure salesforce standard extensions are installed and updated
- Upload Latest vsix salesforcedx-einstein-gpt-0.14.0.vsix
- Make sure salesforce standard extensions are installed and updated
- Upload Latest vsix salesforcedx-vscode-apex-58.4.1.vsix
- Make sure salesforce standard extensions are installed and updated
-  Upload Latest vsix salesforcedx-vscode-core-58.4.1.vsix
- Make sure salesforce standard extensions are installed and updated
- Then on sandbox code builder authenticate with an org with einstein enabled (einstein logo below on the right on visual studio code must be white)
- Create a class and perform the generate code feature

**Comment by samirbavkar** [comment]
Thank you very much for looking into this , [abrantes23](https://github.com/abrantes23).

Yes, I have a 'red' einstein logo , which indicates I do not have permissions to my default org. I am going to try to setup a new developer org and give it a try.  I will post an update.

**Reply by samirbavkar** [reply]
Once I change the Api Client Setting for Einstein GPT for Developers Extension from "Salesforce API "( which was the default) to "BlockCodeGen" I see "SFDX: Generate Code with Einstein GPT"in the Command Palette, however it throws a "SFApiRequestError: eGPT Request Error"

**Reply by samirbavkar** [reply]
I am all set. My main issue was that I was using an incorrect org that was not enabled for Einstein GPT.
